core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java

```diff
@@ -151,8 +151,8 @@ public void close() {
     byte[] startKey = RocksUtils.toByteArray(id, 0);
     byte[] endKey = RocksUtils.toByteArray(id, Long.MAX_VALUE);
 
-    try (RocksIterator iter = db().newIterator(mBlockLocationsColumn.get(),
-        new ReadOptions().setIterateUpperBound(new Slice(endKey)))) {
+    final ReadOptions readOptions = new ReadOptions().setIterateUpperBound(new Slice(endKey));
+    try (RocksIterator iter = db().newIterator(mBlockLocationsColumn.get(), readOptions)) {
```

Comment:
How does this change prevent segsiv? or else why does the original cause one?

Reply:
Read the description.

Reply:
The claim is that because there is no hard reference to ReadOptions in Java. The garbage collector may collect the object away even though the RocksDB native code still needs it.

Reply:
@bradyoo  can you add some javadoc before line 154 to prevent regression by accident in the future ?